lib/oauth.js

```diff
@@ -558,7 +558,7 @@ internals.encode = function (string) {
 internals.parse = function (payload) {
 
     payload = Buffer.isBuffer(payload) ? payload.toString() : payload;
-    if (payload[0] === '{') {
+    if (/^\s*{/.test(payload)) {
```

Comment:
just curious, but is calling `payload.trim()` to remove leading and trailing spaces enough to avoid the regex here?

can anyone think of any edge cases where maintaining leading or trailing whitespace would be necessary?